### PR TITLE
Enable PHP 7.2-8.1 compat

### PR DESCRIPTION
**Pull request by koriym**

### CODE REVIEW

**Comment by harikt** [comment]
Oh very late to the party. I see the PR is on 2021. Sorry missed to notice it.

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         }
     ],
     "require": {
-        "php": ">=5.5.0",
+        "php": "^7.2 || ^8.0",
```
Comment (harikt) [inline composer.json:20]: I would prefer to go with 8.x. You can see 7.x active support already ended. 

image

**Review by harikt — COMMENTED** [review]
I have no other objections in PR other than the version for PHP. iirc @pmjones was also in favour of 8.x for 4.x components. I am going to merge this PR and we can make another PR on the 8.x changes if needed.

Note : Does scrutinizer really work ? I have hard time to make it work.

File: composer.json (hunk 1 above)
Comment (fisharebest) [reply composer.json:20]: > I would prefer to go with 8.x. You can see 7.x active support already ended.

The telemetry from my application (approx 3000 installations) shows the overwhelming majority of sites are still using PHP 7.4

Screenshot 2022-01-26 at 10 29 59

Is there a technical reason to drop support for PHP 7.4?

Aura runs on a very wide range of PHP versions, which makes it great to use in applications like mine, which will be deployed on a wide range of servers.

Could it not be dropped in a later release (e.g. 4.1.)

Comment (harikt) [reply composer.json:20]: We will reconsider this. :) .

Comment (koriym) [reply composer.json:20]: @harikt　Here's what @pmjones had to say.
https://github.com/auraphp/Aura.Cli_Kernel/pull/12#issuecomment-873248524
 It's easy to keep PHP 7.2 to 8.1 compatible with a single version of PHP Unit.

**Comment by koriym** [comment]
> : Does scrutinizer really work ? 

@harikt 

It is no longer necessary to run phpcs or phpmd with scrutinizer (there is GH Actions if needed), and I am not objecting to excluding scrutinizer.

Only scrutinizer's own php analyzer is still valid. It's easy and  working now.

File: composer.json (hunk 1 above)
Comment (harikt) [reply composer.json:20]: Cool

**Comment by koriym** [comment]
@harikt Besides, I think it would be good to demonstrate the high score (10.00 !) thanks to @pmjones' clean code (The real reason I'm keeping it :)

**Pull request merged**